### PR TITLE
LibraryPanels: Prevents deletion of connected library panels

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR prevents the user from deleting a connected/linked library panel both in the Frontend and Backend.

**Library panel linked to dashboard** 
image

**Library not panel linked to dashboard** 
image

**Which issue(s) this PR fixes**:
Relates #31305

**Special notes for your reviewer**:
I considered removing the buttons from the Modal when the library panel is connected/linked. Please let me know what you think. Thank you 🙏

### CODE REVIEW

File: public/app/features/library-panels/components/LibraryPanelsView/actions.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export function searchForLibraryPanels(args: SearchArgs): DispatchResult {
         name: args.searchString,
         perPage: args.perPage,
         page: args.page,
+        excludeUid: args.currentPanelId,
```
Comment (hugohaggmark) [inline public/app/features/library-panels/components/LibraryPanelsView/actions.ts:20]: Fixed an unrelated issue that I found, forgot to add this when I refactored the search previously.

**Review by aknuds1 — APPROVED** [review]

File: pkg/services/librarypanels/database.go [hunk 2]
```diff
@@ -173,10 +173,19 @@ func (lps *LibraryPanelService) deleteLibraryPanel(c *models.ReqContext, uid str
...
+		sql := "SELECT dashboard_id FROM library_panel_dashboard WHERE librarypanel_id=?"
+		if err := session.SQL(sql, panel.ID).Find(&dashIDs); err != nil {
 			return err
+		} else if len(dashIDs) > 0 {
+			return errLibraryPanelHasConnectedDashboards
 		}
 
+		if _, err := session.Exec("DELETE FROM library_panel_dashboard WHERE librarypanel_id=?", panel.ID); err != nil {
+			return err
+		}
```
Comment (torkelo) [inline pkg/services/librarypanels/database.go:188]: Is this needed, the condition above should enforce that this table is empty right?

Comment (hugohaggmark) [reply pkg/services/librarypanels/database.go:188]: Heheh, yes great catch!

Comment (hugohaggmark) [reply pkg/services/librarypanels/database.go:188]: Will fix ASAP

File: public/app/features/library-panels/components/DeleteLibraryPanelModal/DeleteLibraryPanelModal.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  );
+  const asyncDispatch = useMemo(() => asyncDispatcher(dispatch), [dispatch]);
+  useEffect(() => {
+    asyncDispatch(getConnectedDashboards(libraryPanel));
+  }, []);
+  const connected = Boolean(dashboardTitles.length);
+  const done = loadingState === LoadingState.Done;
+
+  return (
+    <Modal className={styles.modal} title="Delete library panel" icon="trash-alt" onDismiss={onDismiss} isOpen={true}>
```
Comment (torkelo) [inline public/app/features/library-panels/components/DeleteLibraryPanelModal/DeleteLibraryPanelModal.tsx:31]: what is these styles.modal & getModalStyles? does the modal not have this built-in?

Comment (hugohaggmark) [reply public/app/features/library-panels/components/DeleteLibraryPanelModal/DeleteLibraryPanelModal.tsx:31]: styles.modal is just for the width, looked super wide without it.

File: pkg/services/librarypanels/database.go (hunk 2 above)
Comment (hugohaggmark) [reply pkg/services/librarypanels/database.go]: Fixed!

File: public/app/features/library-panels/components/DeleteLibraryPanelModal/DeleteLibraryPanelModal.tsx (hunk 3 above)
Comment (torkelo) [reply public/app/features/library-panels/components/DeleteLibraryPanelModal/DeleteLibraryPanelModal.tsx:31]: Ah, I see now itä's only for the width. the getModalStyles made me think it was some global modal styles that had to be passed in.

Comment (hugohaggmark) [reply public/app/features/library-panels/components/DeleteLibraryPanelModal/DeleteLibraryPanelModal.tsx:31]: Yepp, it shared between 2 LibaryPanel modals.

**Review by torkelo — APPROVED** [review]
Think this looks good!

**Pull request merged**